python/tank/bootstrap/manager.py

```diff
@@ -202,30 +208,42 @@ def _set_bundle_cache_fallback_paths(self, paths):
     )
 
 
-    def set_progress_callback(self, callback):
+    def set_progress_callback(self, progress_callback):
         """
-        Specify a method to call whenever progress should be reported back.
+        Sets a function to call whenever progress of the bootstrap should be reported back.
+
+        This function should have the following signature::
 
-        The method needs to have the following signature::
+            progress_callback(progress_value, message, current_index, maximum_index)
 
-            progress_callback(message, current_index, max_index)
+        where:
+        - ``progress_value`` is the current progress value, a float number ranging from 0.0 to 1.0
+                             representing the percentage of work completed.
+        - ``message`` is the progress message string to report.
+        - ``current_index`` is an optional current item number being looped over.
```

Comment:
Simplifying code is good. I integrated into the message the info given by `current_index` and `maximum_index` in the one call that was using it. Gone are the index parameters.